low-level/message_handlers/node_data_msg_handler.py

```diff
@@ -649,13 +701,47 @@ def _generate_if_data(self):
 
             if nw_state == self.FAULT:
                 event_field = f'Network interface {nw_resource_id} is down'
+                if nw_resource_id == pvt_data_nw_if_name:
+                    if not self.is_other_node_reachable:
+                        event_field = f'{other_node}' + ' is not reachable over ' + \
+                                'private data network because of interface fault'
             else:
                 event_field = f'Network interface {nw_resource_id} is up'
+                if nw_resource_id == pvt_data_nw_if_name:
+                    # If the other node is now reachable
+                    if self.is_other_node_reachable:
+                        event_field = f'{other_node}' + ' is reachable over ' + \
+                                'private data network after interface is up'
+                    else:
+                        # We need to send a connection level fault alert
+                        self.pvt_data_nw_conn_unknown_fault = True
+                        # Send conn level alert
+                        event_field_2 = f'{other_node}' + ' is not reachable over ' + \
+                                'private data network because of unknown reason'
+                        severity_2 = self.severity_reader.map_severity(self.FAULT)
+                        self._send_ifdata_json_msg("nw", nw_resource_id, self.NW_RESOURCE_TYPE,
+                                    self.FAULT, severity_2, event_field_2)
 
             # If no or for othe interface, send the alert
             severity = self.severity_reader.map_severity(nw_state)
             self._send_ifdata_json_msg("nw", nw_resource_id, self.NW_RESOURCE_TYPE, nw_state, severity, event_field)
 
+        # Check for conn level faults due to unknown reason

```

Comment:
So this covers private nw connection fault beyond nw port or cable fault rt? Can you add up in code comment here too